app/team.go

```diff
@@ -285,29 +285,31 @@ func (a *App) joinUserToTeam(team *model.Team, user *model.User) (bool, *model.A
 	}
 
 	if etmr := <-a.Srv.Store.Team().GetMember(team.Id, user.Id); etmr.Err == nil {
-		// Membership alredy exists.  Check if deleted and and update, otherwise do nothing
+		// Membership already exists.  Check if deleted and and update, otherwise do nothing
 		rtm := etmr.Data.(*model.TeamMember)
 
 		// Do nothing if already added
 		if rtm.DeleteAt == 0 {
-			return true, nil
+			return rtm, true, nil
 		}
 
 		if tmr := <-a.Srv.Store.Team().UpdateMember(tm); tmr.Err != nil {
-			return false, tmr.Err
+			return nil, false, tmr.Err
+		} else {
+			return tmr.Data.(*model.TeamMember), true, nil
 		}
 	} else {
 		// Membership appears to be missing.  Lets try to add.
 		if tmr := <-a.Srv.Store.Team().SaveMember(tm); tmr.Err != nil {
-			return false, tmr.Err
+			return nil, false, tmr.Err
+		} else {
+			return tmr.Data.(*model.TeamMember), true, nil
```

Comment:
This should be returning `false` right? Since the membership was missing and wasn't already added

Reply:
It was wrong, but not in that way. I've pushed a fix and documented the return values of the function properly.